docs(heartbeat): add browser monitor configuration (#25506)
elastic/beats: heartbeat/docs/monitors/monitor-browser.asciidoc

[[monitor-browser-options]]
=== Browser options

> Also see <<monitor-options>>.

Reviewer: @bmorelli25 can you cross link to https://github.com/elastic/observability-docs/pull/609 ? These reference docs are great, but really, people can't use this without the quickstart.

Reviewer 2: 👍 Added https://github.com/elastic/beats/pull/25506/files#r626119889 with a suggestion to move the link to `<<monitor-options>>` down on the page and instead link to the quick start guide up top.

> Also see <<monitor-options>>.
> 
> The options described here configure {beatname_uc} to run the synthetic
> monitoring test suites via Synthetic Agent on the Chromium browser.
> 

Reviewer 2: ```suggestion
TIP: Want to get started with synthetic monitoring?
See the {observability-guide}/synthetics-quickstart.html[quick start guide].

The options described here configure {beatname_uc} to run the synthetic
monitoring test suites via Synthetic Agent on the Chromium browser.
Additional shared options are defined in <<monitor-options>>.
```

Author: Thanks 👍

[[monitor-browser-options]]
=== Browser options

Also see <<monitor-options>>.

The options described here configure {beatname_uc} to run the synthetic

> monitoring test suites via Synthetic Agent on the chromium browser.

Reviewer 2: ```suggestion
monitoring test suites via Synthetic Agent on the Chromium browser.
```

Example configuration:

[source,yaml]
----
- type: browser
  id: synthetic-inline-suites
  name: Elastic website
  schedule: '@every 1m'
  source:
    inline:
      script: |-
        step("load homepage", async () => {
          await page.goto('https://www.elastic.co');
        });
        step("hover over products menu", async () => {
          await page.hover('css=[data-nav-item=products]');
        });
----

[float]
[[monitor-browser-source]]
==== `source`

Contains information on how to run the synthetic test suites.

[float]
[[monitor-source-inline]]
===== `inline`

Runs the Synthetic test scripts that are defined inline in the configuration.
See {observability-guide}/synthetics-create-test.html[Synthetics syntax] for
more information.

> The mmost convinient way to get the Synthetic test up and running, however it becomes
> harder to maintain and share code once the number of test file increases.
> Recommended way to run Synthetic suites is to create an https://docs.npmjs.com/cli/v7/commands/npm-init[NPM project] and

Reviewer 2: ```suggestion
This is the most convenient way to get Synthetic tests up and running, however, it becomes
harder to maintain and share code as the number of test files increases.
The recommended way to run Synthetic suites is to create an https://docs.npmjs.com/cli/v7/commands/npm-init[NPM project] and
```

[[monitor-browser-options]]
=== Browser options

Also see <<monitor-options>>.

The options described here configure {beatname_uc} to run the synthetic
monitoring test suites via Synthetic Agent on the chromium browser.

Example configuration:

[source,yaml]
----
- type: browser
  id: synthetic-inline-suites
  name: Elastic website
  schedule: '@every 1m'
  source:
    inline:
      script: |-
        step("load homepage", async () => {
          await page.goto('https://www.elastic.co');
        });
        step("hover over products menu", async () => {
          await page.hover('css=[data-nav-item=products]');
        });
----

[float]
[[monitor-browser-source]]
==== `Source`

> Details about the how to run the synthetic test suites.

Reviewer: the wording here is a bit odd. The source field is less about how to run the synthetics test suites then where. Maybe this should say: "Where to find the test to run". @dedemorton thoughts?

[float]
[[monitor-source-inline]]

> ==== `Inline`

Reviewer 2: Nest this under `source`

```suggestion
===== `inline`
```

Runs the Synthetic test scripts that are defined inline in the configuration.

> See <<synthetics-syntax>> for more information.

Author: @bmorelli25 Can we crosslink from here to `observaility-docs`? Interested in this one. -https://www.elastic.co/guide/en/observability/master/synthetics-create-test.html#synthetics-syntax

[float]
[[monitor-source-zipendpoint]]

> ==== `Zip endpoint`

Reviewer: ```suggestion
==== `Zip URL`
```

> ==== `Zip endpoint`

Reviewer 2: Why is this titled `zip endpoint` instead of `zip_url`? Also, we should nest this under `source`
```suggestion
===== `Zip endpoint`
```

> ===== `Zip endpoint`

Author: ```suggestion
===== `Zip URL`
```

Remote zip endpoint configuration allows users to specify the location

> where the synthetics tests are located.

Reviewer: ```suggestion
of a ZIP file where the synthetics tests are located.
```

> Remote zip endpoint configuration allows users to specify the location
> where the synthetics tests are located.

Reviewer 2: ```suggestion
Remote zip endpoint configuration allows users to specify
where the synthetics tests are located.
```

> Under `zip_url`, specify these options

Reviewer 2: ```suggestion
Under `zip_url`, specify these options:
```

*`url`*:: Location of the synthetics project repository.
*`folder`*:: Relative directory path where the synthetic journey files are
located in the repository.
*`username`*:: The username for authenticating with the zip endpoint. This setting is optional.
*`password`*:: The password for authenticating with the zip endpoint. This setting is optional.

> Username and password if provided will be sent as HTTP Basic Authentication

Reviewer 2: ```suggestion
If `username` and `password` are provided, they will be sent as HTTP Basic Authentication
```
